### PR TITLE
Add set_rate to PlayerClock

### PR DESCRIPTION
**Pull request by emersonknapp**
Implements the feature to change playback rate while running

Part of #696

### CODE REVIEW

File: rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp [hunk 1]
```diff
@@ -200,3 +200,31 @@ TEST_F(TimeControllerClockTest, resumes_at_correct_ros_time)
...
   EXPECT_EQ(clock.now(), RCUTILS_S_TO_NS(1));
 }
+
+TEST_F(TimeControllerClockTest, change_rate)
+{
+  rosbag2_cpp::TimeControllerClock clock(ros_start_time, 1.0, now_fn);
+  rcutils_time_point_value_t expected_ros_time = 0;
+
+  // 1 second passes for both
+  return_time += std::chrono::seconds(1);
```
Comment (prajakta-gokhale) [inline rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp:210]: Is `return_time` used anywhere?

Comment (emersonknapp) [reply rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp:210]: yes, `return_time` is the value returned by the mocked `now_fn` that we give to the clock. (https://github.com/ros2/rosbag2/pull/727/files#diff-ea9ff4b69cd29013d4599d8e20313ab1f1dd918e82e810755dad97abfd9f4a51R32) - this is how I am able to control the "flow of steady time" in the tests to get specific scenarios.

Comment (prajakta-gokhale) [reply rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp:210]: That makes sense, thanks!

**Review by prajakta-gokhale — APPROVED** [review]

**Comment by emersonknapp** [comment]
Gist: https://gist.githubusercontent.com/emersonknapp/5009b6ddc40598664a1dd92d52aff43b/raw/8d5251e681f0086ea9df8db8f761ceb096cc92d0/ros2.repos
BUILD args: --packages-up-to rosbag2_cpp
TEST args: --packages-select rosbag2_cpp
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/8223
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14368/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9146/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12046/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14465/)

**Pull request merged**